pubtools/_quay/tag_images.py

```diff
@@ -0,0 +1,176 @@
+    },
+    ("--remote-exec",): {
+        "help": "Flag of whether the commands should be executed on a remote server.",
+        "required": False,
+        "type": bool,
+    },
+    ("--ssh-remote-host",): {
+        "help": "Hostname for remote execution.",
+        "required": False,
+        "type": str,
+    },
+    ("--ssh-remote-host-port",): {
+        "help": "Port of the remote host",
+        "required": False,
+        "type": int,
+    },
+    ("--ssh-reject-unknown-host",): {
+        "help": "Flag of whether to reject an SSH host when it's not found among known hosts.",
+        "required": False,
+        "type": bool,
+    },
+    ("--ssh-username",): {
+        "help": "Username for SSH connection. Defaults to local username.",
+        "required": False,
+        "type": str,
+    },
+    ("--ssh-password",): {
+        "help": "Password for SSH. Will only be used if key-based validation is not available. "
+        "Can be specified by env variable SSH_PASSWORD",
+        "required": False,
+        "type": str,
+        "env_variable": "SSH_PASSWORD",
+    },
+    ("--ssh-key-filename",): {
+        "help": "Path to the private key file for SSH authentication.",
+        "required": False,
+        "type": str,
+    },
+    ("--send-umb-msg",): {
+        "help": "Flag of whether to send a UMB message",
+        "required": False,
+        "type": bool,
+    },
+    ("--umb-url",): {
+        "help": "UMB URL. More than one can be specified.",
+        "required": False,
+        "type": str,
+        "action": "append",
+    },
+    ("--umb-cert",): {
+        "help": "Path to the UMB certificate for SSL authentication.",
+        "required": False,
+        "type": str,
+    },
+    ("--umb-client-key",): {
+        "help": "Path to the UMB private key for accessing the certificate.",
+        "required": False,
+        "type": str,
+    },
+    ("--umb-ca-cert",): {
+        "help": "Path to the UMB CA certificate.",
+        "required": False,
+        "type": str,
+    },
+    ("--umb-topic",): {
+        "help": "UMB topic to send the message to.",
+        "required": False,
+        "type": str,
+    },
+}
+
+
+def tag_images(args):
+    """Tag images main function."""
+    if args.remote_exec:
+        accept_host = (
+            not args.ssh_reject_unknown_host if args.ssh_reject_unknown_host else True
+        )
+        executor = RemoteExecutor(
+            args.ssh_remote_host,
+            args.ssh_username,
+            args.ssh_key_filename,
+            args.ssh_password,
+            args.ssh_remote_host_port,
+            accept_host,
+        )
+    else:
+        executor = LocalExecutor()
+
+    dest_refs = args.dest_ref if isinstance(args.dest_ref, list) else [args.dest_ref]
+    executor.skopeo_login(args.quay_user, args.quay_token)
+    executor.tag_images(args.source_ref, dest_refs)
+
+    if args.send_umb_msg:
+        from rhmsg.activemq.producer import AMQProducer
```

Comment:
can we have this as separate method? It can be later shared with untag entry point.